src/fstab-generator/fstab-generator.c

```diff
@@ -567,6 +577,45 @@ static int add_mount(
                         return r;
         }
 
+        if (units_to_start) {
+                r = strv_consume(units_to_start, TAKE_PTR(automount_name) ?: TAKE_PTR(name));
+                if (r < 0)
+                        return r;
+        }
+
+        return 0;
+}
+
+static int start_units(char **units_to_start) {
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        int r;
+
+        if (!units_to_start) {
+                log_debug("No units to start.");
+                return 0;
+        }
+
+        r = bus_connect_system_systemd(&bus);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get D-Bus connection: %m");
```

Comment:
this is explicitly not OK in a generator. While the generators run PID 1 blocks and just waits for them to complete. This cannot work this way, sorry. We need to find some other way.

See generator man page.

Reply:
No, it's not invoked as a generator. It's effectively a multi-call binary that gets invoked as a generator or as a normal service.